spring-cloud-consul-discovery/src/test/java/org/springframework/cloud/consul/discovery/ConsulDiscoveryClientQueryTagsTests.java

```diff
@@ -0,0 +1,215 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.consul.discovery;
+
+import java.util.Arrays;
+import java.util.List;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+
+import com.ecwid.consul.v1.ConsulClient;
+import com.ecwid.consul.v1.agent.model.NewService;
+import org.junit.jupiter.api.Test;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.AutoConfigurations;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.boot.test.context.assertj.AssertableApplicationContext;
+import org.springframework.boot.test.context.runner.ApplicationContextRunner;
+import org.springframework.cloud.client.ServiceInstance;
+import org.springframework.cloud.consul.test.ConsulTestcontainers;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Import;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Lightweight integration tests to verify Consul query tags usage in
+ * {@link ConsulDiscoveryClient#getInstances(String)}.
+ *
+ * @author Chris Bono
+ */
+class ConsulDiscoveryClientQueryTagsTests {
+
+	private static final String NAME = "query-tags-test-services";
+	static NewService QA_WEST_SERVICE = serviceForEnvironmentAndRegion("qa", "us-west", 9080);
+	static NewService QA_EAST_SERVICE = serviceForEnvironmentAndRegion("qa", "us-east", 9080);
+	static NewService PROD_WEST_SERVICE = serviceForEnvironmentAndRegion("prod", "us-west", 9082);
+	static NewService PROD_EAST_SERVICE = serviceForEnvironmentAndRegion("prod", "us-east", 9082);
+
+	private ApplicationContextRunner appContextRunner = new ApplicationContextRunner()
+			.withInitializer(new ConsulTestcontainers()).withConfiguration(AutoConfigurations.of(TestConfig.class))
+			.withPropertyValues("spring.application.name=consulServiceQueryTags",
+					"spring.cloud.consul.discovery.catalogServicesWatch.enabled=false");
+
+	private static NewService serviceForEnvironmentAndRegion(String env, String region, int port) {
+		NewService service = new NewService();
+		service.setAddress("localhost");
+		service.setId(String.format("%s-%s-%s", NAME, env, region));
+		service.setName(NAME);
+		service.setPort(port);
+		service.setTags(Arrays.asList(env, region));
+		return service;
+	}
+
+	@Test
+	void singleMatchingTagSpecifiedOnDefaultQueryTagProperty() {
```

Comment:
I am using Jupiter rather than Junit4 here. The class/method visibility modifiers reflect that as do the method names (not having to be prefixed w/ `test`)